src/main/kotlin/com/cognifide/gradle/aem/common/instance/service/workflow/WorkflowScheduler.kt

```diff
@@ -0,0 +1,58 @@
+package com.cognifide.gradle.aem.common.instance.service.workflow
+
+import com.cognifide.gradle.aem.common.instance.service.repository.Node
+import org.apache.http.HttpStatus
+
+class WorkflowScheduler(private val workflow: Workflow) {
+
+    private val instance = workflow.instance
+
+    private val logger = workflow.logger
+
+    fun schedule(nodes: Iterable<Node>): Int {
+        nodes.forEach { scheduleForNode(it) }
+        return nodes.count()
```

Comment:
this is not optimal. since you are iterating over iterable once during forEach... you could count there.
if you are calling count() on the iterable you are forcing to iterate over it again which may be expensive.

```kotlin
var count = 0
nodes.forEach { 
  schedule(it) 
  count++
}
return count
```